gandalf.js

```diff
@@ -6,14 +6,16 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.